proto/engine/v1/json_marshal_unmarshal.go

```diff
@@ -1259,3 +1264,21 @@ func RecastHexutilByteSlice(h []hexutil.Bytes) [][]byte {
 	}
 	return r
 }
+
+// UnmarshalJSON implements the json unmarshaler interface for BlobAndProof.
+func (b *BlobAndProof) UnmarshalJSON(enc []byte) error {
+	var dec *BlobAndProofJson
+	if err := json.Unmarshal(enc, &dec); err != nil {
+		return err
+	}
+
+	blob := make([]byte, fieldparams.BlobLength)
```

Comment:
you can use this function `bytesutil.DecodeHexWithLength(dec.Blob, fieldparams.BlobLength)` to make sure there is a length check for dec.Blob

Reply:
I’m sort of leaning against doing it because, at first glance, `DecodeHexWithLength` looks more tedious, and it does an unnecessary round trip of encoding for this usage. We trust the length from EL in many instances here...